### PR TITLE
Add TypeScript definitions

### PR DESCRIPTION
**Pull request by metonym**
- add TypeScript definitions in `index.d.ts`
- adds `types` field in package.json

### CODE REVIEW

**Review by phloe — APPROVED** [review]
Looks good to me 🙂

**Comment by phloe** [comment]
Maybe @voischev has an opinion?

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  attrs: {
+    id?: string;
+    class?: RegExp;
+    [attr: string]: any;
+  };
+}
+
+declare function createMatcher(matcher: string | string[]): Matcher | Matcher[];
+
+export default createMatcher;
```
Comment (Scrum) [inline index.d.ts:12]: there is already a part of types and declarations
https://github.com/posthtml/posthtml-parser/blob/0db4129ec0be89d3278149a4f8140a0c97facd6e/src/index.ts#L16-L28

**Review by Scrum — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (Scrum) [reply index.d.ts:12]: > What's the requested change? To re-use the `NodeTag`?

yep



> One thing to note is that the [`Attributes` type](https://github.com/posthtml/posthtml-parser/blob/0db4129ec0be89d3278149a4f8140a0c97facd6e/src/index.ts#L17) doesn't have `RegExp` as a possible value.

you can always extend the base type

**Comment by metonym** [comment]
@Scrum I noticed that `posthtml` does have a [`StringMatcher` type](https://github.com/posthtml/posthtml/blob/b3fd4ad2dbe8e9a8c0d4b6f9f0f9559a8ded1fc4/types/posthtml.d.ts#L6), which can be a string or `RegExp`.

I updated the typedefs to re-use types from `posthtml`.

Note that I didn't install `posthtml` as a dependency because it's already specified as a peer dependency.

I've also tested the types locally by copying/pasting them here: https://github.com/posthtml/posthtml-insert-at/blob/8ee129cc385a80159fbe843d49df73ba64543f3d/src/global.d.ts

**Review by phloe — APPROVED** [review]

**Comment by phloe** [comment]
LGTM

**Comment by phloe** [comment]
Now available on npm as posthtml-match-helper@1.0.2

**Pull request merged**